pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -47,14 +47,57 @@ type CloudEntitySelectorReconciler struct {
 	Scheme               *runtime.Scheme
 	selectorToAccountMap map[types.NamespacedName]types.NamespacedName
 	Poller               *Poller
+	pendingSyncCount     int
+	Mgr                  *ctrl.Manager
+	initialized          bool
 }
 
 // +kubebuilder:rbac:groups=crd.cloud.antrea.io,resources=cloudentityselectors,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=crd.cloud.antrea.io,resources=cloudentityselectors/status,verbs=get;update;patch
 
+// setPendingSync sets the count of CRs to be reconciled.
+func (r *CloudEntitySelectorReconciler) setPendingSync() error {
+	cesList := &cloudv1alpha1.CloudEntitySelectorList{}
+	if err := r.List(context.TODO(), cesList, &client.ListOptions{}); err != nil {
+		r.Log.Error(err, errorMsgFailedToGetList)
+		return err
+	}
+	r.pendingSyncCount = len(cesList.Items)
+	return nil
+}
+
+// updatePendingSync decreases the count of pending CRs to be reconciled
+// and sets the controller sync status.
+func (r *CloudEntitySelectorReconciler) updatePendingSync() {
+	if r.pendingSyncCount >= 1 {
+		r.pendingSyncCount--
+		// sync completed, update the status.
+		if r.pendingSyncCount == 0 {
+			r.Log.Info(controllerSyncDone)
+			SetControllerSyncStatus(CESController)
+		}
+	}
+}
+
+// waitForControllerInit blocks until controller is initialized.
+func (r *CloudEntitySelectorReconciler) waitForControllerInit() {
+	// Block until controller is initialized.
+	_ = wait.PollInfinite(time.Second*2, func() (bool, error) {
```

Comment:
we need to fail initialization and exit controller.. otherwise nothing will be printed on console

Reply:
Yes, I have added a conservative timeout (can be tweaked) to avoid seeing reconciler errors if a controller is not initialized yet.